rasa/core/exporter.py

```diff
@@ -241,6 +241,11 @@ async def _fetch_events_within_time_range(self) -> AsyncIterator[Dict[Text, Any]
                 continue
 
             events = self._get_events_for_conversation_id(_events, conversation_id)
+
+            # the order of events was changed after ATO-2192
+            # we should sort the events by timestamp to keep the order
+            events.sort(key=lambda x: x["timestamp"])
```

Comment:
certain tests in exporter failed nonetheless so I've added this statement to sort the event list. This sorting does not inflate the performance cost much as it is only used during `rasa export` command -- which is a fire-and-forget command that is seldom used

Reply:
Shouldn't we update the tests instead of adding this extra sort function 🤔

Reply:
The performance may be inflated quite well for large dataset.

Reply:
Ideally this sorting shouldn't be required, we only need this here because `exporter` relies on synthetic data that isn't realistic. However adding sorting here is a way to minimise the 2nd order effects from this change. In terms of performance, sorting the events explicitly here is better than sorting by timestamp in the tracker store query. The tracker store query is used much more frequently compared to `rasa export` command